cmd/podman/volumes/create.go

```diff
@@ -61,6 +66,17 @@ func create(cmd *cobra.Command, args []string) error {
 	)
 	if len(args) > 0 {
 		createOpts.Name = args[0]
+
+		if opts.Ignore {
+			existsResponse, err := registry.ContainerEngine().VolumeExists(context.Background(), createOpts.Name)
```

Comment:
There's a TOCTOU race.

Reply:
There is always a TOCTOU race in any call to "podman create volume", no? We can make it smaller, but nothing keeps the volume alive after registry.ContainerEngine().VolumeCreate() returns.

Reply:
I'd expect the name to be checked and assigned under a lock.

Reply:
Yes, but we return the name with the lock released so there are no guarantees that the volume of that name exists by the time podman prints the name.

Reply:
The correct way is to always create the volume and only if it fails with volume exists ignore that error.

Reply:
Agreed.  And that should happen in the backend as mentioned above.

Reply:
I am changing the code to that, which will make things better. But fundamentally, if you have some process removing volumes without cooperation, there is no race-free way to use the "podman volume create" api (the cli or the rest api).

You are worrying that the ordering is:

```
Task A                              Task B
check if volume exist => yes
                                    remove volume
return old volume name
use the returned volume name => ENOVOLUME
```

Here a lock around the check and create will solve this problem, but there is still this possible ordering:


```
Task A                              Task B
check if volume exist => no
create volume
return new volume name
                                    remove volume
use the returned volume name => ENOVOLUME
```
And no code inside podman can ever fix that.

Reply:
That is true but what it will fix is two podman volume create --ignore name calls at the same time, both can pass the volume exists check before but only one will create it and thus the other one would fail since the error was not checked.